### PR TITLE
incr.comp.: Add `-C incremental` in addition to `-Z incremental`

### PR DESCRIPTION
**Pull request by michaelwoerister**
This PR adds a stable commandline option for invoking incremental compilation.

r? @alexcrichton

### CODE REVIEW

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -1663,7 +1665,24 @@ pub fn build_session_options_and_crate_config(matches: &getopts::Matches)
         early_error(error_format, "Value for codegen units must be a positive nonzero integer");
     }
 
-    if cg.lto && debugging_opts.incremental.is_some() {
+    let incremental = match (&debugging_opts.incremental, &cg.incremental) {
+        (&Some(ref path1), &Some(ref path2)) => {
+            if path1 == path2 {
```
Comment (bjorn3) [inline src/librustc/session/config.rs]: Shouldnt this be a `!=`, currently it errors when there is no conflict

**Review by bjorn3 — CHANGES_REQUESTED** [review]

**Comment by alexcrichton** [comment]
r=me with a typo fix

File: src/librustc/session/config.rs (hunk 1 above)
Comment (michaelwoerister) [reply src/librustc/session/config.rs]: Good catch! Thanks, @bjorn3.

**Comment by MSleepyPanda** [comment]
Might travis be stuck somewhere? The build for i686-gnu [hasn't started yet](https://travis-ci.org/rust-lang/rust/jobs/318654036) but according to travis status there is no backlog for linux

**Comment by alexcrichton** [comment]
@bors: retry

* presumed spurious Travis bug...

**Pull request merged**